pkg/controllers/vdb/sandboxsubcluster_reconciler.go

```diff
@@ -68,9 +72,51 @@ func (s *SandboxSubclusterReconciler) Reconcile(ctx context.Context, _ *ctrl.Req
 		return ctrl.Result{}, err
 	}
 
+	// reconcile sandbox status for the subclusters that are already sandboxed
+	if err := s.reconcileSandboxStatus(ctx); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// reconcile sandbox config maps for the existing sandboxes
+	if err := s.reconcileSandboxConfigMaps(ctx); err != nil {
```

Comment:
I wonder if there will be race conditions by creating the configMap before the actual sandbox command. Once the ConfigMap is created, the Sandbox reconciler can start to act on the pods. What is preventing them from attempting to restart the pods while we are in the middle of sandboxing them. I think this needs to happen after the sandbox command.

Reply:
Here we just create config maps for the sandboxes that have subclusters. This is for the case we sandbox a subcluster successfully but create a config map failed. Then the reconciler will fix the config map here. I have the code to create the configMap after the actual sandbox command as well.

Reply:
Ah ok, yes I see that. You can disregard this comment then.